lucille-core/src/main/java/com/kmwllc/lucille/util/JDBCUtils.java

```diff
@@ -0,0 +1,131 @@
+            doc.setOrAdd(fieldName, integer);
+          }
+          break;
+        case Types.BIGINT:
+          long longVal = rs.getLong(columnIndex);
+          if (!rs.wasNull()) {
+            doc.setOrAdd(fieldName,longVal);
+          }
+          break;
+        case Types.DECIMAL:
+        case Types.NUMERIC:
+        case Types.DOUBLE:
+          double doubleVal = rs.getDouble(columnIndex);
+          // rs.getDouble would return 0 if SQL value is null. This would to check that the field value is not null
+          if (!rs.wasNull()) {
+            doc.setOrAdd(fieldName, doubleVal);
+          }
+          break;
+        case Types.FLOAT:
+        case Types.REAL:
+          float floatVal = rs.getFloat(columnIndex);
+          // rs.getFloat would return 0 if SQL value is null. This would to check that the field value is not null
+          if (!rs.wasNull()) {
+            doc.setOrAdd(fieldName, floatVal);
+          }
+          break;
+        case Types.BOOLEAN:
+        case Types.BIT:
+          boolean boolVal = rs.getBoolean(columnIndex);
+          // rs.getBoolean would return false if SQL value is null. This would to check that the field value is not null
+          if (!rs.wasNull()) {
+            doc.setOrAdd(fieldName, boolVal);
+          }
+          break;
+        case Types.DATE:
+          // Currently saving date types as Strings, as conversion to an Instant may change the value given
+          // Date to Timestamp: sets nanoseconds & time to 0, e.g. 2024-07-30 -> 2024-07-30 00:00:00.0
+          // Timestamp to Instant: checks either DB server local time (which might be from JVM), and converts to UTC,
+          //  e.g. 2024-07-30 00:00:00.0 UTC-4 -> 2024-07-30 04:00:00.0
+          Date dateVal = rs.getDate(columnIndex);
+          if (dateVal != null) {
+            doc.setOrAdd(fieldName, dateVal.toString());
+          }
+          break;
+        case Types.TIMESTAMP:
+          // saving timestamp as str to prevent conversion to UTC
+          // if TIMESTAMP = 2023-01-01T00:00:00Z when local time is UTC-4 in JVM or db,
+          // will be stored as 2023-01-01T04:00:00Z in Document if converted to Instant
+          Timestamp timestamp = rs.getTimestamp(columnIndex);
+          if (timestamp != null) {
+            doc.setOrAdd(fieldName, timestamp.toString());
+          }
+          break;
+        // case Types.TIMESTAMP_WITH_TIMEZONE:
+        // if TIMESTAMP_WITH_TIMEZONE = 2023-01-01T00:00:00-4, will be retrieved as 2023-01-01T04:00:00Z in getTimestamp()
+        // better not save it if it would be converted
+        case Types.BLOB:
+        case Types.BINARY:
+        case Types.VARBINARY:
+        case Types.LONGVARBINARY:
+          byte[] bytes = rs.getBytes(columnIndex);
+          if (bytes != null) {
+            doc.setOrAdd(fieldName, bytes);
+          }
+          break;
+        default:
+          log.warn("SQL Type {} not supported", columnType);
```

Comment:
I think it might be nice to include the actual sql Type name rather than just the number in this log. I think a way to do this would be something like:
`log.warn("SQL Type {} ({}) not supported", JDBCType.valueOf(columnType).getName(), columnType);`
Not super necessary though because we can look up the type easily from the number.
Other than this, everything looks good to me!

Reply:
done!